### PR TITLE
Expose all charting components

### LINKED ISSUE #395: Expose private charting components
- Axis
- AxisLabel
- Bar
- Bars
- Line
- Lines
- Point
- Points

pasted image at 2016_06_28 03_56 pm

### PR DESCRIPTION
**Pull request by jondlm**
## PR Checklist

- Manually tested across supported browsers
  - [x] Chrome
  - [x] Firefox
  - [x] Safari
  - [x] IE11 (Win 7)
  - [x] Edge (Win 10)
- [x] Unit tests written
- [x] PR has one of the `semver-` labels
- [ ] Two core team engineer approvals
- [x] One core team UX approval
- [x] Write unit tests for `Points`

- Fixed a bug with colorOffsets
- Typos in contrib
- Exposing a couple d3 libraries off ours
- Added dashed lines to Line
- Wrote a crap ton of tests
- Upgraded d3 to latest
- Got rid of `top` and `left` in several components in favor of simple `<g transform`s
- Fixed a pie chart pin test

Fixes #395 

http://docspot.devnxs.net/projects/lucid/feature-395-expose-chart-primitives/#/components/Axis
http://docspot.devnxs.net/projects/lucid/feature-395-expose-chart-primitives/#/components/AxisLabel
http://docspot.devnxs.net/projects/lucid/feature-395-expose-chart-primitives/#/components/Bar
http://docspot.devnxs.net/projects/lucid/feature-395-expose-chart-primitives/#/components/Bars
http://docspot.devnxs.net/projects/lucid/feature-395-expose-chart-primitives/#/components/Line
http://docspot.devnxs.net/projects/lucid/feature-395-expose-chart-primitives/#/components/Lines
http://docspot.devnxs.net/projects/lucid/feature-395-expose-chart-primitives/#/components/Point
http://docspot.devnxs.net/projects/lucid/feature-395-expose-chart-primitives/#/components/Points

### CODE REVIEW

File: src/components/Line/Line.jsx [hunk 1]
```diff
@@ -75,6 +83,7 @@ const Line = createClass({
 				}}
```
Comment (sodiumjoe) [inline src/components/Line/Line.jsx]: should probably use `omitProps` here and elsewhere

Comment (jondlm) [inline src/components/Line/Line.jsx]: Definitely. I'll fix that up!

Comment (jondlm) [inline src/components/Line/Line.jsx]: Looks like you already did that in #414 -- correct me if I'm wrong.

Comment (sodiumjoe) [inline src/components/Line/Line.jsx]: ah, yes, you're right, i forgot these aren't new files. There are no new component files in this PR?

**Pull request merged**